### PR TITLE
js-basics-diagnostic

### PR DESCRIPTION
**Pull request by treep78**
Comfort: 4
Clarity: 4

I completed the diagnostic completely from memory.

### CODE REVIEW

**Comment by treep78** [comment]
I have been stuck down by my own hubris!

File: bin/diagnostic.js [hunk 1]
```diff
@@ -1 +1,27 @@
...
+const ask = require('../lib/ask.js');
+
+let answer = 'not empty';
+
+while (answer !== '')
+{
+  answer = ask("Guess my secret? ");
+  if(answer > 0 && answer <= 10)
+  {
+    if (answer === 8)
```
Comment (neugierige) [inline bin/diagnostic.js:13]: This never evaluates to `true`. Do you know why that might be?

File: diagnostic.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ Assume all code executes in strict mode.
 ## Question 1
 
 Please list the JavaScript primitives and give one example of each.
+Number: 5
+String: "This is a string!"
+Boolean: true
+Object: {key: value}
+Array: ["an", "array", "of", "strings"]
```
Comment (neugierige) [inline diagnostic.md:15]: Arrays and Objects are not actually primitives. Please take a look at the JS Basics lesson again: https://github.com/ga-wdi-boston/js-basics

File: bin/diagnostic.js [hunk 3]
```diff
@@ -1 +1,27 @@
...
+    else
+    {
+      console.log("Nope! Try again!");
+    }
+  }
+  if(answer < 1 || answer > 10)
+  {
+    console.log("You need to choose a number between 1 and 10.");
+  }
+}
```
Comment (neugierige) [inline bin/diagnostic.js:27]: Very close! Always test out your code to see if it's behaving according to your expectations. Here's 1 possible solution: 

```
const ask = require('../lib/ask.js');

let answer = '';

while (answer !== '5') {
  if (answer < 1 || answer > 10) {
   console.log('Please try a number between 1 and 10');
 } else if (answer !== '') {
   console.log('That\'s not it!');
 }
  answer = ask("Guess my secret? ");
}
console.log("You guessed it!");
```

**Comment by neugierige** [comment]
3/5. Hi Peter, I'm glad to hear that you felt confident enough to complete this diagnostic from memory, however, in real life, it is much more important to write code that is works, rather than writing code from memory. If you want to truly test yourself, you can still do that, but maybe double check your responses before submitting!

File: bin/diagnostic.js (hunk 1 above)
Comment (treep78) [reply bin/diagnostic.js:13]: because it should ask either ' == 8 ' or ' ==="8" '

Comment (neugierige) [reply bin/diagnostic.js:13]: 👍

**Pull request closed without merging**